### PR TITLE
Change dependency:  ngx-jodit instead of jodit-angular

### LINKED ISSUE #735: Change Angular Jodit dependency
Our Jodit HTML editor currently depends on the `jodit-angular` package:  

https://www.npmjs.com/package/jodit-angular

This has some dependency conflicts when we move to angular 14 - mostly around the rxjs package.

We should upgrade to `ngx-jodit`

https://www.npmjs.com/package/ngx-jodit

Which has the same functionality but is updated a little more frequently.

### PR DESCRIPTION
**Pull request by jamesrwelch**
Fixes #735

### CODE REVIEW

**Comment by pjmonks** [comment]
image

I can't seem to switch back to `develop` branch due to `npm install` changing a lot of things on my local build, so can't check this myself. But did this custom toolbar button originally have an icon? I can't remember if it existed before or it was a change I made to the pending PR #572

**Comment by pjmonks** [comment]
I see the following TypeScript warnings now:

```
Warning: C:/Code/Mauro/MauroDataMapper/mdm-ui/node_modules/jodit/src/config.ts is part of the TypeScript compilation but it's unused.
Add only entry points to the 'files' or 'include' properties in your tsconfig.

Warning: C:/Code/Mauro/MauroDataMapper/mdm-ui/node_modules/jodit/src/core/constants.ts is part of the TypeScript compilation but it's unused.
Add only entry points to the 'files' or 'include' properties in your tsconfig.

Warning: C:/Code/Mauro/MauroDataMapper/mdm-ui/node_modules/jodit/src/types/storage.ts is part of the TypeScript compilation but it's unused.
Add only entry points to the 'files' or 'include' properties in your tsconfig.

Warning: C:/Code/Mauro/MauroDataMapper/mdm-ui/node_modules/jodit/types/types/storage.ts is part of the TypeScript compilation but it's unused.
Add only entry points to the 'files' or 'include' properties in your tsconfig.
```

I think maybe the `tsconfig.app.json` file could have an `excludes` list added with these files to remove the warnings, but not sure. See https://stackoverflow.com/questions/57729518/how-to-get-rid-of-the-warning-ts-file-is-part-of-the-typescript-compilation-but for more information perhaps.

**Comment by pjmonks** [comment]
Found a bug when editing existing HTML content:

1. Pick a catalogue item and edit the description. Set the value through the HTML editor and save.
2. Confirm the description was saved - the catalogue item should reload and you'll see your changes.
3. Now edit the description again. The HTML editor is blank.
4. Switch the Markdown editor option though and the content is there.

Some sort of binding issue it seems, perhaps `[(ngModel)]` isn't supported in this now Jodit wrapper?

**Review by pjmonks — CHANGES_REQUESTED** [review]
The code changes look fine but found some functional issues when testing. Could you review them please?

**Comment by jamesrwelch** [comment]
> image
> 
> I can't seem to switch back to `develop` branch due to `npm install` changing a lot of things on my local build, so can't check this myself. But did this custom toolbar button originally have an icon? I can't remember if it existed before or it was a change I made to the pending PR #572

You're right - they've changed the way icons work in Jodit.  I couldn't find one I liked in their default set, so I've pushed a commit with a new SVG and a change to code.

File: package.json [hunk 1]
```diff
@@ -60,8 +60,7 @@
     "diff-match-patch": "^1.0.5",
     "dmn-js": "^10.1.0",
     "graphlib": "2.1.8",
-    "jodit": "^3.5.4",
-    "jodit-angular": "^1.12.3",
+    "jodit": "^3.24.2",
```
[the remark this answers is not in the document]
Comment (jamesrwelch) [reply package.json]: I've read this: https://docs.renovatebot.com/dependency-pinning/.  It all makes sense... I think you're right, we should go with their suggestion and pin dependencies.  I've pushed a commit to do this

**Comment by jamesrwelch** [comment]
@pjmonks Thank you so much for your thorough review - I'm glad we're picking up these issues that would have been difficult to spot amongst a much bigger upgrade.  
I believe I've addressed all your issues, so please feel free to re-test.  However, although `ng serve` is working for me, I'm getting a build error running `npm run build`.  I'll look into this next - would be interesting to hear if you have the same problem.  In bouncing between branches I've been finding it helpful to do `npm cache verify` / `npm cache clear --false` in order to get clean installs on a different branch.  I'll be glad when I can stop doing that!

**Comment by jamesrwelch** [comment]
>  I'm getting a build error running `npm run build`. I'll look into this next - would be interesting to hear if you have the same problem. In bouncing between branches I've been finding it helpful to do `npm cache verify` / `npm cache clear --false` in order to get clean installs on a different branch. I'll be glad when I can stop doing that!

I've done some more investigation:
- This only occurs during `npm run build` and not building `npm run dist`, so it's not a huge problem.  It relates to the source maps for debugging TS code.
- It happens for me building develop (recent, and an older version), so it's not a regression
- I don't get the errors in my upgrade to Angular 14, which will be my next PR after this.

So I think we might be ready for re-testing on this one 👍

**Review by pjmonks — APPROVED** [review]
Re-tested and the changes are functioning better now, looks no different using the new Jodit wrapper compared to before 👍 

Approved ✔️

**Pull request merged**